analytics/upress_reporting/models/pollers/download_data_poller.py

```diff
@@ -23,28 +23,17 @@ def __init__(self, *args):
         self.setup_db_manager()
 
         self.set_items()
-        self.set_events()
-
+        self.set_events(self.bucket_name, self.log_path)
```

Comment:
Instantiating a DonwloadDataPoller has a side effect of setting the events. 

You may want to consider changing `set_events` to `get_events` and have the code that creates a Poller object explicitly call `get_events`.

Reply:
Ah okay, wouldn't we want the events to be set before building the report, though? The [`set_events`](https://github.com/NYPL/drb-etl-pipeline/blob/7b4be143ecb3656c5e5bb76502577ed4f91592b2/analytics/upress_reporting/models/aggregators/download_data_aggregator.py#L30) function is simply calling the [`pull_interaction_events`](https://github.com/NYPL/drb-etl-pipeline/blob/7b4be143ecb3656c5e5bb76502577ed4f91592b2/analytics/upress_reporting/models/aggregators/aggregator.py#L36) method but with error handling, i.e. following the same pattern as the previous iterations of this code.

Reply:
Yes we would want to instantiate then make the call to get the events. I'll leave it up to you - no biggie!